client/client_test.go

```diff
@@ -549,7 +595,25 @@ func TestConcurrentIncrements(t *testing.T) {
 		if err := db.DelRange(testUser+"/value-0", testUser+"/value-1x"); err != nil {
 			t.Fatalf("%d: unable to clean up: %v", k, err)
 		}
-		concurrentIncrements(db, t)
+		concurrentIncrements(db, t, false /*isMultiphase*/)
+	}
+}
+
+// TestMultiphaseTransactions is a simple explicit test for serializability
+// while running multiphase trasactions.
+func TestMultiphaseTransaction(t *testing.T) {
+	defer leaktest.AfterTest(t)
+	s := server.StartTestServer(t)
+	defer s.Stop()
+	db := createTestClient(s.ServingAddr())
+
+	// Convenience loop: Crank up this number for testing this
+	// more often. It'll increase test duration though.
+	for k := 0; k < 5; k++ {
+		if err := db.DelRange(testUser+"/value-0", testUser+"/value-1x"); err != nil {
+			t.Fatalf("%d: unable to clean up: %v", k, err)
+		}
+		concurrentIncrements(db, t, true /*isMultiphase*/)
```

Comment:
I'll do that too. Let's see how that plays out.

On Wed, Aug 12, 2015 at 8:58 PM Tamir Duberstein <notifications@github.com>
wrote:

> In client/client_test.go
> <https://github.com/cockroachdb/cockroach/pull/2076#discussion_r36931971>:
>
> > +
> > +// TestMultiphaseTransactions is a simple explicit test for serializability
> > +// while running multiphase trasactions.
> > +func TestMultiphaseTransaction(t *testing.T) {
> > +	defer leaktest.AfterTest(t)
> > +	s := server.StartTestServer(t)
> > +	defer s.Stop()
> > +	db := createTestClient(s.ServingAddr())
> > +
> > +	// Convenience loop: Crank up this number for testing this
> > +	// more often. It'll increase test duration though.
> > +	for k := 0; k < 5; k++ {
> > +		if err := db.DelRange(testUser+"/value-0", testUser+"/value-1x"); err != nil {
> > +			t.Fatalf("%d: unable to clean up: %v", k, err)
> > +		}
> > +		concurrentIncrements(db, t, true /*isMultiphase*/)
>
> Sure, but why is it inappropriate to test that at the driver level?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cockroachdb/cockroach/pull/2076/files#r36931971>.
>